### PR TITLE
Get/delete endpoint

### PR DESCRIPTION
**Pull request by Jypaa**
Added endpoint for get user by id, get all users and delete user by id.

### CODE REVIEW

**Review by JuhoBjn — CHANGES_REQUESTED** [review]
I left some comments for you to check out.

File: backend/models/users.js [hunk 1]
```diff
@@ -1,6 +1,41 @@
...
+  },
+  /**
+   * Find users based on id and delete it.
+   * @param {string} id The user's id
+   * @returns User account found with id
+   */
+  delete: async (id) => {
+    console.log("poistettava id", id);
+    const deleteQuery = "DELETE FROM users WHERE id=?;";
+    await promisePool.query(deleteQuery, [id], (err) => {
```
[the remark this answers is not in the document]
Comment (Jypaa) [reply backend/models/users.js:33]: If I take it off then it tries to run other tests while it is still deleting user.

File: backend/controllers/users.js [hunk 2]
```diff
@@ -2,9 +2,63 @@ const Joi = require("joi");
...
+};
+
+const deleteUser = async (req, res) => {
+  const schema = Joi.object({
+    id: Joi.string().required()
+  });
+  const providedCredentials = {
+    id: req.params.id
+  };
+
```
Comment (vaasuu) [inline backend/controllers/users.js:46]: Add check: User should only be able to delete their own account.
Currently anyone can delete someone else's account!

File: backend/controllers/users.js [hunk 3]
```diff
@@ -2,9 +2,63 @@ const Joi = require("joi");
...
+  } else {
+    res.status(404).json({ message: "No user found with given id" });
+  }
+};
+
+const allUsers = async (req, res) => {
+  console.log(req.params);
+  const response = await userModels.findAll();
+  if (response) {
+    res.status(200).json(response);
```
Comment (vaasuu) [inline backend/controllers/users.js:33]: Do not return everything (including password hashes!), only some things like id, name, postcode, city ?

File: backend/controllers/users.js [hunk 4]
```diff
@@ -2,9 +2,63 @@ const Joi = require("joi");
 const bcrypt = require("bcryptjs");
 const jwt = require("jsonwebtoken");
 const { v4: genUuid } = require("uuid");
-
 const userModels = require("../models/users");
 
+const user = async (req, res) => {
+  const schema = Joi.object({
+    id: Joi.string().required()
```
Comment (vaasuu) [inline backend/controllers/users.js:9]: Id type should be uuid

File: backend/controllers/users.js [hunk 5]
```diff
@@ -2,9 +2,63 @@ const Joi = require("joi");
...
+
+  try {
+    const { error } = schema.validate(providedCredentials);
+    if (error) throw error;
+  } catch (error) {
+    return res.status(400).send({ message: error.details[0].message });
+  }
+  const response = await userModels.findById(providedCredentials.id);
+  if (response) {
+    res.status(200).json(response);
```
Comment (vaasuu) [inline backend/controllers/users.js:23]: Do not return everything (like password hash).

**Review by vaasuu — CHANGES_REQUESTED** [review]
I left some comments

File: backend/controllers/users.js [hunk 6]
```diff
@@ -2,9 +2,63 @@ const Joi = require("joi");
...
+  } catch (error) {
+    return res.status(400).send({ message: error.details[0].message });
+  }
+
+  const response = await userModels.delete(providedCredentials.id);
+  if (response) {
+    res.status(404).json({ message: "user found with given id" });
+  } else {
+    res.status(200).json({ message: "No user found with given id" });
+  }
```
[the remark this answers is not in the document]
Comment (Jypaa) [reply backend/controllers/users.js:59]: I was thinking that if it gets response which it should not get, then throw error code, but maybe 404 is not right one.

**Review by JuhoBjn — APPROVED** [review]

**Review by vaasuu — APPROVED** [review]

**Pull request merged**